src/lem/slot.rs

```diff
@@ -266,14 +277,7 @@ impl SlotType {
     }
 
     pub(crate) fn is_compatible(&self, slot_data: &SlotData) -> bool {
-        matches!(
-            (self, slot_data),
-            (Self::Hash4, SlotData::PtrVec(..))
-                | (Self::Hash6, SlotData::PtrVec(..))
-                | (Self::Hash8, SlotData::PtrVec(..))
-                | (Self::Commitment, SlotData::FPtr(..))
-                | (Self::BitDecomp, SlotData::F(..))
-        )
+        slot_data.size() == self.preimg_size()
```

Comment:
I know this isn't this PR's fault, but can you add some documentation explaining what "being compatible" means in this context, please?